### PR TITLE
nixos/taskserver: Enable cfg.group to read clients' certificates

### LINKED ISSUE #235833: taskserver-init: Fix permissions / allow to configure permissions for `nixos-taskserver` generated client certificates
So I today I spawned a machine with `services.taskserver.enable = true;` and satisfyingly the `/var/lib/taskserver` directory was generated with all of the certificates etc that taskwarrior needs, including the clients' certificates.

I noticed that the users' keys located at `/var/lib/taskserver/keys/<org-name>/<user-name>/` are owned by root with read only permissions for the root user alone. No matter what are `cfg.user` and `cfg.group`. I'd like to enable the group `syncthing` read those client keys, as well as the `ca.cert` file so that I can configure a client easily. Thankfully the `syncthing` group _can_ read the `ca.cert` file, but not the client keys.

The manual indeed explains that the command:

```
ssh server nixos-taskserver user export my-company alice | sh
```

Could be used to get those keys, but I'd like to use syncthing for that to automate the process even further.

I got a bit lost while reading the `helper-tool.py`. Specifically, I'm not sure why does `certtool_cmd` always works in the following umask:

https://github.com/NixOS/nixpkgs/blob/fc6855d83a2c37dafa75298d3b968dfd0cd89597/nixos/modules/services/misc/taskserver/helper-tool.py#L81-L93

I mean, ~why not~ - a solution is suggested in https://github.com/NixOS/nixpkgs/pull/235847 .

A word of comment will be appreciated. cc @aszlig .

### PR DESCRIPTION
**Pull request by doronbehar**
###### Description of changes

This enables the services.taskserver.group to read the certificates generated by the taskserver.service' preStart script.

This needs review.

###### Things done

Tested this on my server - by deleting the `/var/lib/taskserver` directory and then letting the service recreate it by itself and insepcting the permissions afterwards.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aszlig** [comment]
@ofborg test taskserver

File: nixos/modules/services/misc/taskserver/helper-tool.py
Comment (aszlig) [inline nixos/modules/services/misc/taskserver/helper-tool.py:93]: Does that imply that setting the `umask` doesn't work here?

**Comment by aszlig** [comment]
NixOS test is currently broken, fix is in #235972.

File: nixos/modules/services/misc/taskserver/helper-tool.py
Comment (doronbehar) [reply nixos/modules/services/misc/taskserver/helper-tool.py:93]: > Does that imply that setting the `umask` doesn't work here (eg. `certtool` ignores it)?

Pretty much yes. `certtool` ensures by itself that only the user running `certtool` will have read and write access to the private keys that are generated. Hence the `chmod` for adding group read access to the private key is needed.

Comment (aszlig) [reply nixos/modules/services/misc/taskserver/helper-tool.py:93]: Ah, okay... seems reasonable then to do the `chmod`. I'm however never used the `taskserver` module myself (stopped using taskwarrior directly afterwards) even though I implemented it, so maybe I'm not the best one to review this properly. From looking at it however the change looks fine to me.

Comment (doronbehar) [reply nixos/modules/services/misc/taskserver/helper-tool.py:93]: TBH, I don't use taskserver myself as well anymore. I just tested the NixOS module for a brief period and it did work for me during that period. Since this PR doesn't change anything fundamental, I think it'd be safe enough to merge it. But it's your call. We can also ask for a review in one of the discourse threads [1](https://discourse.nixos.org/t/prs-in-distress/3604/60) or [2](https://discourse.nixos.org/t/prs-ready-for-review/3032/485).

Comment (aszlig) [reply nixos/modules/services/misc/taskserver/helper-tool.py:93]: Maybe @arcnmx, @pacien or @Pamplemousse can help there, since they have made changes to the module in recent years. If noone objects I'd merge in a few days since I don't think this change will cause a big disruption.

Comment (Pamplemousse) [reply nixos/modules/services/misc/taskserver/helper-tool.py:93]: I sadly don't have time to test this, but LGTM :+1:

**Comment by doronbehar** [comment]
@aszlig maybe we can merge this now?

**Comment by doronbehar** [comment]
Due to [inthe.am shutting down](https://github.com/coddingtonbear/inthe.am/discussions/427), I started setting up a taskserver on my day-to-day laptop, and I had to make a few more modifications to the `helper-tool.py` script to make all directories leading the keys' files executable, and readable by the group as well. Here's the result I reached which I'm satisfied with:

```
[drwxrwx--- taskd    taskd-group]  /var/lib/taskserver
├── [-rw------- taskd    taskd-group]  config
├── [-rw-r--r-- taskd    taskd-group]  .is_initialized
├── [drwxr-x--- taskd    taskd-group]  keys
│   ├── [-rw-r----- root     taskd-group]  ca.cert
│   ├── [-rw------- root     root    ]  ca.key
│   ├── [drwxr-x--- taskd    taskd-group]  doronbehar
│   │   └── [drwxr-x--- taskd    taskd-group]  doron
│   │       ├── [-rw-r----- root     taskd-group]  private.key
│   │       └── [-rw-r--r-- root     taskd-group]  public.cert
│   ├── [-rw-r----- root     taskd-group]  server.cert
│   ├── [-rw-r----- root     taskd-group]  server.crl
│   └── [-rw-r----- root     taskd-group]  server.key
├── [drwx------ taskd    taskd-group]  orgs
│   └── [drwx------ taskd    taskd-group]  doronbehar
│       ├── [drwx------ taskd    taskd-group]  groups
│       └── [drwx------ taskd    taskd-group]  users
│           └── [drwx------ taskd    taskd-group]  c3daa87a-fcad-446c-a437-f94d806e35df
│               └── [-rw------- taskd    taskd-group]  config
└── [drwxr-xr-x taskd-group taskd-group]  .stfolder
```

I feel confident to merge this and become responsible to any consequences. I also verified that the `nixosTests.taskserver` didn't break.

**Review by aszlig — APPROVED** [review]
> @aszlig maybe we can merge this now?

LGTM from my part, sorry for the late reply.

**Comment by doronbehar** [comment]
@ofborg build nixosTests.taskserver

**Pull request merged**
